src/components/trashlookup.jsx

```diff
@@ -1,156 +1,157 @@
-import React, { Component } from "react";
+import React from "react";
+import Form from "./common/Form";
 import PostData from "../Data/street.json";
-import SearchBox from "./searchBox";
+import { Button } from 'reactstrap';
 import _ from "lodash";
-import ListGroup from "./listGroup";
-import Trash from "./common/trash";
-import Leaf from "./common/leaf";
-import Recycle from "./common/recycle";
+
+
 import { getTrashService } from "../services/trashService";
+import * as moment from "moment";
+import ScheduleTable from './common/scheduleTable';
+class TrashLookUp extends Form {
+ 
+  displayText() {
+    const selectedAddress = _.trim(this.state.value);
+    const linkMessage = "Not the right address?";
+      let message = [];
+      message.push(<h6>Your Schedule</h6>);
+      message.push("showing collection schedule for:");
+      message.push(<h6> {selectedAddress} </h6>);
+      message.push("Not the right address?Try another search");
+   
+
+      const displayMessage = message.map((data, index) => {
+        return <span key={index}>{data}</span>;
+      });
+
+    return(
+     
+<div>
+   {displayMessage} <Button color="link" onClick={this.handleClick}>Try another search</Button>
+   </div>               
+    );
+  }
+
+  getAddresses() {
+    let searchQuery =  _.trim(this.state.value);
+    let filtered =PostData;
+    const isHidden = this.state.isHidden;
+ 
+    if (searchQuery.length > 0) {
+   
+         if (isHidden) 
+         {
+          filtered = PostData.filter(m =>
+            m.address1.toLowerCase().indexOf(searchQuery.toLowerCase()) > -1);
+          }
+          else{
+            filtered = PostData.filter(
+             m => m.address1.toLowerCase() === searchQuery.toLowerCase());
+      
+          }
+        }
+    return {  data: filtered };
+  }
+
+   renderConditionTable() {
+     const dowloadMessage =
+       "Download your complete four year scheduled in PDF format";
+
+    return (
+      <div className="row">
+        <div className="col-5">
+        <ScheduleTable services = {getTrashService()} renderWeekOfDay= {this.renderWeekOfDay} />
+          <h6>Download</h6>
+          {/* <a href="/test">{dowloadMessage}</a> */}
+        </div>
+      </div>
+    );
+ }
+  renderNotFoundMessage() {
+  
+    const searchValue = this.state.data["searchboxname"];
+    return (
+      <div>
+        <i>we could not find address {searchValue}</i>
+      </div>
+    );
+  }
+
+  renderWeekOfDay = type =>  {
+    return  type === 1
+       ? (this.trashNextCollectionDate())
+       : type === 2
+       ? (this.recylceNextCollectionDate())
+       : (this.leafNextCollectionDate());
+
+ 
+  }
+  leafNextCollectionDate() {
+
+    const dayOfWeek = moment().day();
+    return dayOfWeek === 5
+      ? moment()
+          .add(14, "d")
+          .format("D/M/YYYY")
```

Comment:
you should store this format in a variable somewhere, since you use it over and over.

something like

```js
const dateFormat = 'D/M/YYYY';
```

that way if you need to change it, you only need to do it in one place.